core/tests/upgrade-test/tests/upgrade.test.ts

```diff
@@ -50,16 +42,68 @@ describe('Upgrade test', function () {
     let forceDeployBytecode: string;
     let logs: fs.WriteStream;
 
+    let ethProviderAddress: string | undefined;
+    let web3JsonRpc: string | undefined;
+    let contractsL2DefaultUpgradeAddr: string;
+    let deployerAddress: string;
+    let complexUpgraderAddress: string;
+    let upgradeAddress: string | undefined;
+    let contractsPriorityTxMaxGasLimit: string;
+
     before('Create test wallet', async () => {
-        tester = await Tester.init(process.env.CHAIN_ETH_NETWORK || 'localhost');
+        forceDeployAddress = '0xf04ce00000000000000000000000000000000000';
+        deployerAddress = '0x0000000000000000000000000000000000008007';
+        complexUpgraderAddress = '0x000000000000000000000000000000000000800f';
+
+        if (fileConfig.loadFromFile) {
+            const generalConfig = loadConfig({ pathToHome, chain: fileConfig.chain, config: 'general.yaml' });
+            const contractsConfig = loadConfig({ pathToHome, chain: fileConfig.chain, config: 'contracts.yaml' });
+            const secretsConfig = loadConfig({ pathToHome, chain: fileConfig.chain, config: 'secrets.yaml' });
+
+            ethProviderAddress = secretsConfig.l1.l1_rpc_url;
+            web3JsonRpc = generalConfig.api.web3_json_rpc.http_url;
+            contractsL2DefaultUpgradeAddr = contractsConfig.l2.default_l2_upgrader;
+            upgradeAddress = contractsConfig.l1.default_upgrade_addr;
+            contractsPriorityTxMaxGasLimit = '72000000';
+        } else {
+            ethProviderAddress = process.env.L1_RPC_ADDRESS || process.env.ETH_CLIENT_WEB3_URL;
+            web3JsonRpc = process.env.ZKSYNC_WEB3_API_URL || process.env.API_WEB3_JSON_RPC_HTTP_URL;
+            contractsL2DefaultUpgradeAddr = process.env.CONTRACTS_L2_DEFAULT_UPGRADE_ADDR!;
+
+            upgradeAddress = process.env.CONTRACTS_DEFAULT_UPGRADE_ADDR;
+            if (!upgradeAddress) {
+                throw new Error('CONTRACTS_DEFAULT_UPGRADE_ADDR not set');
+            }
+            contractsPriorityTxMaxGasLimit = process.env.CONTRACTS_PRIORITY_TX_MAX_GAS_LIMIT!;
+        }
+
+        const network = process.env.CHAIN_ETH_NETWORK || 'localhost';
+        tester = await Tester.init(network, ethProviderAddress!, web3JsonRpc!);
```

Comment:
let's slightly change the approach with `network` don't pass this variable and just check the existence of `MASTER_WALLET_PK` inside tester and apply the corresponding configs

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/2515/commits/9f0d1af6c01e433ccae43fc3965b0994ccbd8fc2